addons/csi-openstack-cinder/openstack-csi-cinder.yaml

```diff
@@ -0,0 +1,567 @@
+          resources:
+            {}
+        - name: csi-resizer
+          image: {{ .InternalImages.Get "OpenstackCSIResizer" }}
+          imagePullPolicy: IfNotPresent
+          args:
+            - "-v=1"
+            - "--csi-address=$(ADDRESS)"
+            - "--timeout=3m"
+            - "--handle-volume-inuse-error=false"
+            - "--leader-election=true"
+          env:
+            - name: ADDRESS
+              value: /var/lib/csi/sockets/pluginproxy/csi.sock
+          volumeMounts:
+            - name: socket-dir
+              mountPath: /var/lib/csi/sockets/pluginproxy/
+          resources:
+            {}
+        - name: liveness-probe
+          image: {{ .InternalImages.Get "OpenstackCSILivenessProbe" }}
+          imagePullPolicy: IfNotPresent
+          args:
+            - "-v=1"
+            - "--csi-address=$(ADDRESS)"
+          env:
+            - name: ADDRESS
+              value: /var/lib/csi/sockets/pluginproxy/csi.sock
+          volumeMounts:
+            - mountPath: /var/lib/csi/sockets/pluginproxy/
+              name: socket-dir
+          resources:
+            {}
+        - name: cinder-csi-plugin
+          image: {{ .InternalImages.Get "OpenstackCSI" }}
+          imagePullPolicy: IfNotPresent
+          args:
+            - /bin/cinder-csi-plugin
+            - "-v=1"
+            - "--endpoint=$(CSI_ENDPOINT)"
+            - "--cloud-config=$(CLOUD_CONFIG)"
+            - "--cluster=$(CLUSTER_NAME)"
+          env:
+            - name: CSI_ENDPOINT
+              value: unix://csi/csi.sock
+            - name: CLOUD_CONFIG
+              value: /etc/kubernetes/cloud-config
+            - name: CLUSTER_NAME
+              value: "{{ .Config.Name }}"
+{{ if .Config.CABundle }}
+{{ caBundleEnvVar | indent 12 }}
+{{ end }}
+          ports:
+            - containerPort: 9808
+              name: healthz
+              protocol: TCP
+          # The probe
+          livenessProbe:
+            failureThreshold: 5
+            httpGet:
+              path: /healthz
+              port: healthz
+            initialDelaySeconds: 10
+            timeoutSeconds: 10
+            periodSeconds: 60
+          volumeMounts:
+            - name: socket-dir
+              mountPath: /csi
+            - mountPath: /etc/cacert
+              name: cacert
+              readOnly: true
+            - mountPath: /etc/kubernetes
+              name: cloud-config
+              readOnly: true
+{{ if .Config.CABundle }}
+{{ caBundleVolumeMount | indent 12 }}
+{{ end }}
+          resources:
+            {}
+      volumes:
+        - name: socket-dir
+          emptyDir:
+        - name: cloud-config
+          secret:
+            secretName: cloud-config
+
+        - hostPath:
+            path: /etc/cacert
+          name: cacert
+{{ if .Config.CABundle }}
+{{ caBundleVolume | indent 8 }}
+{{ end }}
+      affinity:
+        {}
+      nodeSelector:
+        {}
+      tolerations:
+        []
```

Comment:
Please make sure that nodeSelector and tolerations match what we already have in place.

Reply:
done